plugin/documents.py

```diff
@@ -319,6 +321,11 @@ def on_text_changed_async(self, change_count: int, changes: Iterable[sublime.Tex
         self._when_selection_remains_stable_async(self._do_code_lenses_async, current_region,
                                                   after_ms=self.code_lenses_debounce_time)
 
+        for change in changes:
+            if len(change.str) == 1 and (change.str == ' ' or change.str == ';' or change.str == '.'):
+                self._when_selection_remains_stable_async(self._do_semantic_tokens_async, current_region,
+                                                          after_ms=self.semantic_tokens_debounce_time)
```

Comment:
What's the reasoning for this logic here? Why do we need to trigger semantic tokens on those changes and not the any any other and is this logic universal to all languages?

Reply:
basically the idea is to only trigger semantic tokens after typing a whole word, otherwise sometimes it freaks out and draws incorrectly, especially while holding `backspace` or `return` keys